services/headless-lms/models/src/peer_review_question_submissions.rs

```diff
@@ -41,6 +41,39 @@ RETURNING id
     Ok(res.id)
 }
 
+pub async fn get_by_peer_reviews_question_ids(
+    conn: &mut PgConnection,
+    ids: &[Uuid],
+    user_id: Uuid,
+) -> ModelResult<Vec<PeerReviewQuestionSubmission>> {
+    let res = sqlx::query_as!(
+        PeerReviewQuestionSubmission,
+        "
+    SELECT qs.id,
+        qs.created_at,
+        qs.updated_at,
+        qs.deleted_at,
+        qs.peer_review_question_id,
+        qs.peer_review_submission_id,
+        qs.text_data,
+        qs.number_data
+    FROM peer_review_question_submissions qs
+        JOIN peer_review_submissions s ON (qs.peer_review_submission_id = s.id)
+        JOIN exercise_slide_submissions es ON (s.exercise_slide_submission_id = es.id)
+    WHERE peer_review_question_id IN (
+        SELECT UNNEST($1::uuid [])
+    )
+        AND es.user_id = $2
+        AND qs.deleted_at IS NULL;
```

Comment:
I think this would return also peer review question submissions that have been given to other answers by this user. The other answers could be from another exercises, another course instances, another courses, or they even could be from older submissions to the same exercise.

I think in order to make sure we don't show extra peer review question submissions, we should make this function to take an exercise slide submission id, and only return the peer review question submissions that have been given to the specified exercise slide submission. 

To make obtaining the exercise slide submission id easier, maybe we could change the endpoint to take the exercise slide submission id directly instead of the exercise id i.e. `/api/v0/course-material/exercises/:exercise_id/peer-review-given` -> `/api/v0/course-material/exercise-slide-submissions/:exercise_slide_submission_id/peer-reviews-received`. I would imagine that the exercise slide submission id would be available in the frontend where the endpoint is used.